pkg/apis/security/validation/workloadidentity.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	gardencorevalidation "github.com/gardener/gardener/pkg/apis/core/validation"
+	"github.com/gardener/gardener/pkg/apis/security"
+)
+
+// GetSubClaimPrefixAndDelimiterFunc is func providing the prefix value for the 'sub' claim
+// and the delimiter used to concatenate the various parts.
+var GetSubClaimPrefixAndDelimiterFunc = func() (string, string) {
+	d := ":"
+	p := strings.Join([]string{"gardener.cloud", "workloadidentity"}, d)
+	return p, d
+}
+
+// ValidateWorkloadIdentity validates a WorkloadIdentity.
+func ValidateWorkloadIdentity(workloadIdentity *security.WorkloadIdentity) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&workloadIdentity.ObjectMeta, true, gardencorevalidation.ValidateName, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateSpec(workloadIdentity.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, validateSubClaim(*workloadIdentity, GetSubClaimPrefixAndDelimiterFunc, field.NewPath("status").Child("sub"))...)
+
+	return allErrs
+}
+
+func validateSpec(spec security.WorkloadIdentitySpec, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateAudiences(spec.Audiences, path.Child("audiences"))...)
+	allErrs = append(allErrs, validateTargetSystem(spec.TargetSystem, path.Child("targetSystem"))...)
+
+	return allErrs
+}
+
+// validateAudiences validates a WorkloadIdentity audiences object.
+func validateAudiences(audiences []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(audiences) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, "must provide at least one audience"))
+	}
+
+	for idx, aud := range audiences {
```

Comment:
Does it make sense to check for duplicates?

Reply:
OK, your call. It was just a thought that popped up when I read this code. I'm fine either way.

Reply:
Yeah, IMO duplicate audiences does not make sense. I also vote for implementing the check now since we can always loosen later on.